integration-tests/test/test_unbond.py

```diff
@@ -0,0 +1,58 @@
+from random import Random
+import logging
+from docker.client import DockerClient
+
+from rchain.crypto import PrivateKey
+from . import conftest
+from .common import (
+    CommandLineOptions,
+)
+from .rnode import (
+    bootstrap_connected_peer,
+    extract_validator_stake_from_bonds_validator_str,
+    started_bootstrap,
+    docker_network
+)
+from .wait import (
+    wait_for_node_sees_block,
+)
+
+
+BOOTSTRAP_KEY = PrivateKey.from_hex("b2527b00340a83e302beae2a8daf6d654e8e57541acfa261cc1b5635eb16aa15")
+VALIDATOR_KEY_1 = PrivateKey.from_hex("9a801debae8bb97fe54c99389cafa576c60612503348578125b65ab182ff5850")
+
+def test_unbond_validator(command_line_options: CommandLineOptions, random_generator: Random, docker_client: DockerClient) -> None:
+    bonded_validator_map = {
+        BOOTSTRAP_KEY: 200,
+        VALIDATOR_KEY_1: 300,
+    }
+
+    with conftest.testing_context(command_line_options, random_generator, docker_client, bootstrap_key=BOOTSTRAP_KEY, validator_bonds_dict=bonded_validator_map) as context, \
+        docker_network(context, context.docker) as network, \
+        conftest.temporary_wallets_file(random_generator) as wallets_file, \
+        started_bootstrap(context=context, network=network,mount_dir=context.mount_dir, wallets_file=wallets_file) as bootstrap_node, \
+        bootstrap_connected_peer(context=context, bootstrap=bootstrap_node, name='bonded-validator-1', private_key=VALIDATOR_KEY_1) as validator_1:
+
+        validator_1.deploy('/opt/docker/examples/tut-hello.rho', VALIDATOR_KEY_1)
+        validator_1.propose()
+
+        block_hash = validator_1.deploy_contract_with_substitution(substitute_dict={}, rho_file_path="resources/wallets/unbond.rho", private_key=VALIDATOR_KEY_1)
+
+        wait_for_node_sees_block(context, bootstrap_node, block_hash)
```

Comment:
This waiting uses shell to start a new instance of node to make gRPC request to checking node.
https://github.com/rchain/rchain/blob/5351708/integration-tests/test/rnode.py#L263

This seems super slow. Can we use gRPC directly or events API to check for a block? Although this is probably a change for another ticket to get rid of shell calls in rnode.py.

Reply:
Yes. We can do that.

Reply:
@zsluedem has the change proposed by @tgrospic been made?

Reply:
no

Reply:
@tgrospic @9rb I will try to use some other ways(meybe websocket) to do that kind of things. I admit right now the way we do is not really elegant.